webui/src/lib/remark-plugins/imageUriReplacer.ts

```diff
@@ -0,0 +1,33 @@
+import { visit } from "unist-util-visit";
+import type { Node } from "unist";
+import type { Root } from "mdast";
+import type { Plugin } from "unified";
+
+import { objects } from "../api";
+
+const imageUriReplacer: Plugin<[], Root> = () => async (tree) => {
+  const images: Array<Node & { url: string }> = [];
+  visit(tree, "image", (node: Node & { url: string }) => {
+    if (node.url.startsWith("lakefs://")) {
+      images.push(node);
+    }
+  });
+
+  const promises: Array<Promise<void>> = [];
+  for (const image of images) {
+    const [repo, branch, ...path] = image.url.split("/").slice(2);
+    const promise = objects
+      .getPresignedUrl(repo, branch, path.join("/"))
```

Comment:
What if presigned URLs is not enabled for the installation?

Reply:
Like you can still read the images even if presign urls are not available, right?

Reply:
@itaiad200 This assumes pre-signed URLs are enabled. If not, I could add an alternative to download the file and convert it to a base64 data URL. Do you think that's needed?

Reply:
How can it assumes presigned is enabled if some installations (local, mem) don't support that at all and in other (Azure) it depends on the configuration? 
We have a way to check a installation/repo for presigned support (as it's part of the settings). We can add the images support for repos without presigned urls later, as long as we handle these repos nicely for now, i.e. don't try to embed images for now.

Reply:
I see... I did not consider that... 🤔
I'll see if I can construct data URLs to support installations without presigned URLs.
We also need to know if presigned URLs are available on the client
Maybe it's time to add a "supported features" or "configuration" endpoint so the client is aware of what's available/supported/enabled/disabled/etc.

Reply:
Replaced presigned URLs with "regular" URLs. This also simplifies several other things.